# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webidl/Range.webidl
# Commit: b1f1e6ad93f8
# Full Hash: b1f1e6ad93f8d7634acfcd0d66467c6efc2ad80e
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2024-03-26 04:11:24
# Regressor Bug: 1867058
# File Overlap Count: 1
# Description:
#   Bug 1867058 - Part 5: Update nsRange to create and update mCrossBoundaryRange when necessary r=smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D195305
# ==============================================================================

diff -r 02d89b880ac5 -r b1f1e6ad93f8 dom/webidl/Range.webidl
--- a/dom/webidl/Range.webidl	Mon Mar 25 13:40:57 2024 +0000
+++ b/dom/webidl/Range.webidl	Mon Mar 25 13:40:57 2024 +0000
@@ -94,3 +94,12 @@
   [ChromeOnly, Throws]
   ClientRectsAndTexts getClientRectsAndTexts();
 };
+
+// ChromeOnly methods that allow setting Range boundaries to cross
+// shadow boundary.
+partial interface Range {
+  [ChromeOnly, Throws]
+  undefined setStartAllowCrossShadowBoundary(Node refNode, unsigned long offset);
+  [ChromeOnly, Throws]
+  undefined setEndAllowCrossShadowBoundary(Node refNode, unsigned long offset);
+};
